servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl

```diff
@@ -397,13 +397,13 @@ void main() {
 	mat4 model_matrix = instances.data[instance_index].transform;
 #if defined(MOTION_VECTORS)
 	global_time = scene_data_block.prev_data.time;
-	vertex_shader(instance_index, is_multimesh, scene_data_block.prev_data, instances.data[instance_index].prev_transform, prev_screen_position);
+	vertex_shader(instance_index, is_multimesh, draw_call.multimesh_motion_vectors_previous_offset, scene_data_block.prev_data, instances.data[instance_index].prev_transform, prev_screen_position);
 	global_time = scene_data_block.data.time;
-	vertex_shader(instance_index, is_multimesh, scene_data_block.data, model_matrix, screen_position);
+	vertex_shader(instance_index, is_multimesh, draw_call.multimesh_motion_vectors_current_offset, scene_data_block.data, model_matrix, screen_position);
 #else
 	global_time = scene_data_block.data.time;
 	vec4 screen_position;
-	vertex_shader(instance_index, is_multimesh, scene_data_block.data, model_matrix, screen_position);
+	vertex_shader(instance_index, is_multimesh, draw_call.multimesh_motion_vectors_current_offset, scene_data_block.data, model_matrix, screen_position);
```

Comment:
I wonder if there would be a performance improvement to use ``0`` here instead of ``draw_call.multimesh_motion_vectors_current_offset`` I have a feeling the compiler may be able to compile out ``multimesh_offset`` if it will always be 0

Reply:
Running a draw pass without motion vectors doesn't imply that the current offset i 0. 

The motion vector logic needs to run for all viewports that render motion vectors, but you could have a separate viewport that doesn't do MV. In that case, the viewport will run the non-motion vector branch, but it still needs to use the correct "current offset".

We could eventually add a branch that hard-codes 0 when none of the viewports render motion vectors, but we need to check if it actually makes a difference in performance, and whether the gain is worth the added complexity.

Reply:
Alright, that makes sense. Thanks for clarifying!